test/cocotb/osdtestlib/debug_interconnect.py

```diff
@@ -33,23 +59,97 @@ def transmit_flit(self, dut, data, last):
             yield RisingEdge(dut.clk)
             dut.debug_in.valid <= 0
 
+        dut.debug_in.last <= 0
+
 
 class DiPacket:
     """
     Represents a complete debug interconnect packet
     """
 
-    def __init__(self, destination, source, packet_type, packet_subtype, 
-                 payload):
+    def __init__(self, destination=1, source=0,
+                 packet_type=PACKET_TYPE_REG,
+                 packet_subtype=PACKET_SUBTYPE_REQ_READ_REG_16,
+                 payload=[]):
 
-        self.destination    = destination
-        self.source         = source
-        self.packet_type    = packet_type
+        self.new = 1
+
+        self.destination = destination
+        self.source = source
+        self.packet_type = packet_type
         self.packet_subtype = packet_subtype
```

Comment:
You can rename those two from packet_subtype and packet_type to just type and subtype. If you access a DiPacket.type field, it's clear that we're talking about the packet type. Additionally, this naming makes it consistent with the destination and source fields (which, btw, should be named dest and src to keep them consistent with the documentation http://opensocdebug.readthedocs.io/en/latest/02_spec/03_dataformats.html.)